File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.js
const inorderTraversal = (root) => {
    const values = [];
    const stack = [];

    let current = root;

    while (current !== null || stack.length !== 0) {
        if (current !== null) {
          stack.push(current);
          current = current.left;
        } else {
          const last = stack.pop();
          values.push(last.val);
          current = last.right;
        }
    }

    // while (current !== null || stack.length !== 0) {
    //     while (current !== null) {
    //         stack.push(current);
    //         current = current.left;
    //     }

    //     const last = stack.pop();
    //     values.push(last.val);
    //     current = last.right
    // }

    return values;
};

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
class Solution:
    def inorderTraversal(self, root):
        values = []
        stack = []

        current = root

        while current is not None or len(stack) > 0:
            if current is not None:
                stack.append(current)
                current = current.left
            else:
                last = stack.pop()
                values.append(last.val)
                current = last.right

        return values
